source/val/validate_extensions.cpp

```diff
@@ -2028,6 +2131,377 @@ spv_result_t ValidateExtInst(ValidationState_t& _, const Instruction* inst) {
         break;
       }
     }
+  } else if (ext_inst_type == SPV_EXT_INST_TYPE_OPENCL_DEBUGINFO_100) {
+    if (!_.IsVoidType(result_type)) {
+      return _.diag(SPV_ERROR_INVALID_DATA, inst)
+             << ext_inst_name() << ": "
+             << "expected result type must be a result id of "
+             << "OpTypeVoid";
+    }
+
+    const OpenCLDebugInfo100Instructions ext_inst_key =
+        OpenCLDebugInfo100Instructions(ext_inst_index);
+    switch (ext_inst_key) {
+      case OpenCLDebugInfo100DebugInfoNone:
+      case OpenCLDebugInfo100DebugNoScope:
+      case OpenCLDebugInfo100DebugOperation:
+        break;
+      case OpenCLDebugInfo100DebugCompilationUnit: {
+        auto validate_src = ValidateDebugInfoOperand(
+            _, "Source", OpenCLDebugInfo100DebugSource, inst, 7, ext_inst_name);
+        if (validate_src != SPV_SUCCESS) return validate_src;
+        break;
+      }
+      case OpenCLDebugInfo100DebugSource: {
+        auto validate_file = ValidateOperandForDebugInfo(
+            _, "File", SpvOpString, inst, 5, ext_inst_name);
+        if (validate_file != SPV_SUCCESS) return validate_file;
+        if (num_operands == 6) {
```

Comment:
This is a subtle problem.  This code uses num_operands, which is  inst->operands().size().
But then the helper function accesses inst->words(index).
The two vectors operands_ and words_ are different things. It's a happy accident that they have the same number of elements.  But it's confusing to use the count of one to control the indexing of the other.

If you're going to be accessing the word vector, then get a num_words variable from inst->words().size(), and use num_words consistently below.

The operands_ member if Instruction is not the same as the words